### PR TITLE
Harden publish-dist

### PR DESCRIPTION
**Pull request by martinpitt**
This avoids all recently discussed attack vectors of our build-dist/publish-dist workflow pair.

I tested this on my fork, for both a [push](https://github.com/martinpitt/cockpit-machines/runs/2476801243?check_suite_focus=true) and a [pull_request](https://github.com/martinpitt/cockpit-machines/runs/2476801243?check_suite_focus=true) from https://github.com/martinpitt/cockpit-machines/pull/4.

In both cases publishing succeeded, committed the expected sha-TAG to the -dist repo, and I validated that the cache works with `GITHUB_BASE=martinpitt/cockpit-machines test/download-dist`.

 - [x] After landing this, re-enable `pull_request` webhook

### CODE REVIEW

**Comment by martinpitt** [comment]
Note that publish-dist for *this PR* (and hence packit) **cannot** work -- this PR drops the SHA from the artifact, but publish-dist runs from main and expects this file. Hence the testing on my fork.

**Comment by martinpitt** [comment]
I triggered fedora-33 manually, as that's a required context. As explained above, the others cannot be triggered here.

**Comment by martinpitt** [comment]
I added a commit for eliminating events.json as well. Unfortunately there is no way to get trustworthy information about the original  pull_request event, so we have to give up on this experiment.

**Comment by martinpitt** [comment]
@marusak , @KKoukiou : As Lis is out today again, can you please review this? Should be pretty straightforward, and it's the things that Lis and I already talked about, and her review in https://github.com/cockpit-project/cockpit/pull/15748

File: .github/workflows/publish-dist.yml [hunk 1]
```diff
@@ -28,42 +29,16 @@ jobs:
...
           mkdir dist-repo
           cd dist-repo
           git init
           cp -r ../dist/ ../package-lock.json .
 
           git add .
-          git commit -m "Build for $SHA"
-          tag="sha-$SHA"
+          git commit -m "Build for ${{ github.event.workflow_run.head_sha }}"
+          tag="sha-${{ github.event.workflow_run.head_sha }}"
```
Comment (allisonkarlitskaya) [inline .github/workflows/publish-dist.yml:40]: I think you can theoretically switch to single-quotes here.

**Review by allisonkarlitskaya — APPROVED** [review]

File: .github/workflows/publish-dist.yml (hunk 1 above)
Comment (martinpitt) [reply .github/workflows/publish-dist.yml:40]: Yes, and I already did that in the cockpit PR. There are a few other trivialities which I want to get in sync, I'll do that in a follow-up.

**Pull request merged**